video/out/vo_sixel.c

```diff
@@ -296,6 +299,10 @@ static void flip_page(struct vo *vo)
 {
     struct priv* priv = vo->priv;
 
+    // Make sure that image and dither are valid before drawing
+    if (priv->buffer == NULL || priv->dither == NULL)
```

Comment:
You never set it to NULL, so you can't count on it being NULL initially. You should probably NULL the whole priv struct at `preinit`.

Reply:
Actually in `preinit` `priv->buffer` is assigned using `talloc_array` at the end before draw_image is ever called. `priv->dither` is also set to NULL over there

Reply:
Actually, my bad. it's initialized implicitly at `video_out_sixel = { ...` because it's missing there. So no need for explicit init to NULL.

So just make the titles shorter please.

Reply:
For reference, clarification of the init process (first time I've looked at vo init):
- The driver defines `struct vo_driver` with functions, and optionally initializes `priv_defaults` to point to the initial value of priv, together with `priv_size` which holds its size.
- During init, mpv allocates `priv_size` and points the `priv` member of `struct vo` to this allocation, and then copies into it the data from `priv_defaults` of `struct vo_driver`.

Therefore, the initial value of `priv->output` and `priv->dither` of the `struct vo` pointer on entry to the functions (e.g. `page_flip`) is from `priv_defaults` of `struct vo_driver`.

And because `.priv_defaults = &(const struct priv) { ...` doesn't explicitly initialize `.output` and `.priv` - they're initialized to NULL.